partner_delivery_zone/models/sale_order.py

```diff
@@ -17,3 +17,16 @@ class SaleOrder(models.Model):
     def onchange_partner_shipping_id_delivery_zone(self):
         if self.partner_shipping_id.delivery_zone_id:
             self.delivery_zone_id = self.partner_shipping_id.delivery_zone_id
+
+    def write(self, vals):
+        # Update picking delivery zone if user update it in sale order that
+        # creates a picking,
+        res = super().write(vals)
+        if ('delivery_zone_id' in vals and not
```

Comment:
IMO `delivery_zone_id` shouldn't be updateable in sales order once you have confirmed the sales order.